providers/gce/gce_loadbalancer.go

```diff
@@ -137,6 +137,13 @@ func (g *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, svc
 		return nil, err
 	}
 
+	// GCE load balancers do not support services with LoadBalancerClass set. LoadBalancerClass can't be updated for an existing load balancer, so here we don't need to clean any resources.
+	// Check API documentation for .Spec.LoadBalancerClass for details on when this field is allowed to be changed.
+	if svc.Spec.LoadBalancerClass != nil {
```

Comment:
should not it be at the top, first thing to check? if is a customer loadbalancer cluster the `g.ClusterID.GetID()` seems unnnecessary

Reply:
moved up, above the mixed protos check. Would be bad to report a mixed protos error for a service we don't intend to provision an LB for.

Reply:
I was suggesting to move it at the top, first thing of anything

Reply:
moved all the way up